Update root level readme

luca-della-vedova opened the pull request:
This PR adds a package description to the top level readme, including a high level overview of the repository organization and details about the supported Gazebo versions and the update schedule.

Yadunund commented on README.md:
```diff
@@ -1 +1,22 @@
+
+#### Repository organization
+
+Most plugins are split into a `common`, `gz` and `gz_classic` package. The `common` package contains the simulator independent logic, such as ROS 2 publishers / subscribers, core logic, while the `gz` and `gz_classic` packages contain a wrapper to interface to the specific simulators.
```
```suggestion
Most plugins are split into a `common`, `gz` and `gz_classic` package. The `common` package contains the simulator independent logic, such as ROS 2 publishers / subscriptions, core logic, while the `gz` and `gz_classic` packages contain a wrapper to interface to the specific simulators.
```

Yadunund commented on README.md:
```diff
@@ -1 +1,22 @@
+
+#### Supported versions
+
+Since Gazebo and ROS 2 have a different release schedule, this package will follow the [ros_gz](https://github.com/gazebosim/ros_gz) supported version and, depending on which ROS 2 version RMF is currently targeting, the Gazebo version will be set accordingly to allow using the binary release of the `ros_gz_bridge`.
```
```suggestion
Since Gazebo and ROS 2 have a different release schedule, this package will follow the [ros_gz](https://github.com/gazebosim/ros_gz) supported version and, depending on which ROS 2 version Open-RMF is currently targeting, the Gazebo version will be set accordingly to allow using the binary release of the `ros_gz_bridge`.
```

Yadunund commented on README.md:
```diff
@@ -1 +1,22 @@
+
+1. Building simulation plugins under `rmf_building_sim_*`, are used to simulate different aspect of buildings such as doors, lifts and crowds (currently through the [Menge library](https://github.com/open-rmf/menge_vendor)). Utility plugins to help with simulation, such as UI widgets to toggle battery consumption and set charging behavior, as well as toggling floor visibility, are also placed in  ...
+
+1. Robot simulation plugins under `rmf_robot_sim_*` are used to simulate aspects of robots behavior.
```
```suggestion
2. Robot simulation plugins under `rmf_robot_sim_*` are used to simulate aspects of robots behavior.
```

Yadunund commented on README.md:
```diff
@@ -1 +1,22 @@
+
+Simulation plugins are split into two categories:
+
+1. Building simulation plugins under `rmf_building_sim_*`, are used to simulate different aspect of buildings such as doors, lifts and crowds (currently through the [Menge library](https://github.com/open-rmf/menge_vendor)). Utility plugins to help with simulation, such as UI widgets to toggle battery consumption and set charging behavior, as well as toggling floor visibility, are also placed in  ...
```
Would be a bit nicer if each plugin name had its own indented bullet with an explanation. Also good if we can highlight (in a table?) the name of each plugin, whether its a world or model plugin, and the SDF schema when including this plugin eg door 
```
      <plugin name="door" filename="libdoor.so">
        <v_max_door>0.5</v_max_door>
        <a_max_door>0.3</a_max_door>
        <a_nom_door>0.15</a_nom_door>
        <dx_min_door>0.01</dx_min_door>
        <f_max_door>500.0</f_max_door>
        <door name="hardware_door" type="SwingDoor" left_joint_name="empty_joint" right_joint_name="right_joint" />
      </plugin>
```

Bonus if we can document each parameter in the plugins

Yadunund reviewed (changes requested):
Minor feedback

luca-della-vedova commented:
I added a more detailed explanation of each plugin, what it does, as well as an SDF snippet and parameter explanation https://github.com/open-rmf/rmf_simulation/pull/100/commits/8d06999364609245588a846ffdbf7f2877799d09. I believe I also addressed the remaining points.

I looked into having the documentation in the code and pointing to it similar to what other repos (i.e. `rmf_battery`) do, but these packages use the SDF, rather than the C++ code, as a public API so the underlying implementation can change while SDF snippets should still work as intended.

Yadunund reviewed (approved):
Nice
